src/main/java/com/palantir/docker/compose/execution/DockerComposeExecutable.java

```diff
@@ -32,37 +34,39 @@
             "/usr/local/bin/docker-compose"
     );
 
-    private final DockerComposeFiles dockerComposeFiles;
-    private final DockerConfiguration dockerConfiguration;
-    private final String dockerComposePath;
+    @Value.Parameter protected abstract DockerComposeFiles dockerComposeFiles();
+    @Value.Parameter protected abstract DockerConfiguration dockerConfiguration();
+
+    @Value.Default protected DockerComposeProjectName projectName() {
+        return new DockerComposeProjectName();
```

Comment:
I know this already merged, but I'd be much happier if this was DockerComposeProjectName.random()